### PR TITLE
Add ABOUT modal window

### PR DESCRIPTION
**Pull request by janper**
A very annoying thing happened! We won't be mentioned as authors of this software at the exhibition at MAK. I'm totally mad and angry! It's partly due to SCD not acknowledging our authorship in general (keeps perpetually happening with the entire exhibition too) and significantly also because we (or rather I personally) didn't check the texts before printing them out. 

**PLEASE ACCEPT MY DEEPEST APOLOGY!**

I'm trying to make it up as much as I can (and know) with this PR. I wish I knew how to display images with IMGUI, but I don't. In such case I'd put screenshots, logos and nicely formatted text (as an image) into the modal window, as well as Subdigital logo into the corner of the viewport.
For now it's just this. 

Please, with this PR don't only check the technicalities, but also the contents and also let me know how do you want to be mentioned.

I'm also considering turning on the About window on the program startup.

This is the text that will be printed on the table ext to the computer at the MAK exhibition in Vienna:
```
H.U.R.B.A.N. selector

AutorInnen
Maroš Schmidt (Konzept)
Lucia Dubačová
(Projektkoordination/Forschungs- und Entwicklungsabteilung des Slovak Design Center)
Subdigital (Entwicklung)

Die Software wurde im Rahmen des Projekts Design & Innovation, Grenzüberschreitende 
Zusammenarbeit zwischen Designinstitutionen im Umbruch zum digitalen Zeitalter entwickelt, 
das vom Kooperationsprogramm Interreg V-A Slowakei - Österreich 2014-2020 unterstützt wird.

Die Software H.U.R.B.A.N. selector ist Teil des Vorhabens der Forschungs- und 
Entwicklungsabteilung des Slovak Design Center – Inolab, eine Forschungsplattform 
zu schaffen, auf der DesignerInnen neue Algorithmen testen bzw. verifizieren und neue 
Formen erstellen können. 
H.U.R.B.A.N. selector ermöglicht die Hybridisierung mehrerer 3D-Modelle, um neue 
ästhetische Formen zu finden. Es wurde für DesignerInnen zum Experimentieren entwickelt 
und ist ein erster Schritt in Richtung vollwertige parametrische Entwurfssoftware. Der 
Benutzer erstellt eine Folge von Vorgängen, die er jederzeit neu konfigurieren kann, 
was sich wiederum auf die In- und Outputs anderer Vorgänge auswirkt. 
Das Programm erweitert die kreativen Möglichkeiten von Designern/Designerinnen und 
hilft ihnen, jenseits ihrer Fähigkeiten und der Grenzen ihrer Vorstellungskraft, die durch die
Gedächtnisleistung gegeben ist, stets neue Formvariationen und Kompositionen zu kreieren. 
Dies ermöglicht ihnen, sich mehr auf die Selektion und die algorithmische Suche nach neuen 
Formen zu konzentrieren.

HURBAN selector befindet sich derzeit in der Entwicklungsphase. Die Software ist in Rust 
programmiert und der Quellcode ist frei verfügbar. 




The H.U.R.B.A.N. Selector software is a part of the SDC’s Inolab Department’s plan to create a 
research platform for designers to test and verify new algorithms and create new forms.

H.U.R.B.A.N. Selector is a software which allows hybridization of multiple 3D models with the 
target to find new aesthetic forms. It is intended for designers as a tool for experimenting. 
Therefore, it acts as a gateway to full-fledged parametric design software.
A user builds an operation pipeline where each stacked operation allows reconfiguration 
anytime influencing inputs and outputs of subsequent operations.

The program extends the creative possibilities of designers and helps them create beyond 
the limits of their imagination given by memory/brain capacity as well as the ability to create 
different variations of form and compositions. Using H.U.R.B.A.N. Selector provides designers 
with more space for selection and algorithmic seeking of new forms. 
 
H.U.R.B.A.N. Selector is currently under development. It utilizes the Rust programming 
language, and its source code is freely available to the community.

```

### CODE REVIEW

**Review by ondrowan — DISMISSED** [review]

**Comment by yanchith** [comment]
I'll add placeholder pictures and leave this open for you :)

**Comment by yanchith** [comment]
Added examples on how to add images. Have a look and modify the about page to your liking. The text seems good to me.

**Review by yanchith — DISMISSED** [review]

File: src/ui.rs [hunk 1]
```diff
@@ -478,29 +499,31 @@ impl<'a> UiFrame<'a> {
...
             .always_auto_resize(true)
             .build(|| {
+                let window_width = ui.window_size()[0];
+
                 let wrap_token = ui.push_text_wrap_pos(ABOUT_WINDOW_WIDTH);
-                ui.text(imgui::im_str!("H.U.R.B.A.N. selector by Subdigital s.r.o. (https://sub.digital), 2020\n\
-                \n"));
+
+                let big_bold_font_token = ui.push_font(self.font_ids.big_bold);
+                ui.text(imgui::im_str!("H.U.R.B.A.N. selector 1.0 beta"));
```
Comment (yanchith) [inline src/ui.rs]: Do we really need to be so cocky and call this a 1.0 beta? I would call this 0.1 alpha at most. Can we meet somewhere in the middle?

File: src/ui.rs [hunk 2]
```diff
@@ -459,12 +472,20 @@ impl<'a> UiFrame<'a> {
     pub fn draw_about_window(
         &self,
         about_modal_open: &mut bool,
-        tex_janci: imgui::TextureId,
-        tex_janper: imgui::TextureId,
-        tex_ondro: imgui::TextureId,
+        tex_scheme: imgui::TextureId,
+        width_scheme: f32,
```
Comment (yanchith) [inline src/ui.rs]: Make these u32 and cast in functions in UI. Applies to all dimensions you added in the PR

File: src/ui.rs [hunk 3]
```diff
@@ -459,12 +472,20 @@ impl<'a> UiFrame<'a> {
...
+        tex_logos: imgui::TextureId,
+        width_logos: f32,
+        height_logos: f32,
     ) {
         let ui = &self.imgui_ui;
 
+        let window_color_token = ui.push_style_color(
+            imgui::StyleColor::PopupBg,
+            self.colors.popup_window_background,
+        );
```
Comment (yanchith) [inline src/ui.rs:488]: Note that this makes the credits quite unreadable in the dark theme.

File: src/ui.rs [hunk 4]
```diff
@@ -641,6 +673,63 @@ impl<'a> UiFrame<'a> {
...
+                    subdigital_logo_window_vertical_position,
+                ],
+                imgui::Condition::Always,
+            )
+            .build(ui, || {
+                imgui::Image::new(
+                    tex_subdigital_logo,
+                    [SUBDIGITAL_LOGO_WINDOW_WIDTH, subdigital_logo_window_height],
+                )
+                .build(ui);
```
Comment (yanchith) [inline src/ui.rs:729]: This image is almost invisible in the funky theme. Maybe we want to make a dark version of the logo, similarly to the logos in the about window?

**Review by yanchith — COMMENTED** [review]
See inline

File: src/ui.rs (hunk 1 above)
Comment (janper) [reply src/ui.rs]: I don't mind. 0.1 alpha it is!

File: src/ui.rs (hunk 3 above)
Comment (janper) [reply src/ui.rs:488]: I'd keep it default but there is no variable storing the default window color - or at leas not that I know. I can change it to something else though.

File: src/ui.rs (hunk 4 above)
Comment (janper) [reply src/ui.rs:729]: it's almost intentional. I'll check some other possibilities. maybe 50% grey would work for both.

**Review by yanchith — APPROVED** [review]

**Pull request merged**